src/Storages/MergeTree/GinIndexStore.cpp

```diff
@@ -511,6 +513,30 @@ void GinIndexStore::cancel() noexcept
         postings_file_stream->cancel();
 }
 
+String GinIndexStore::Statistics::toString() const
+{
+    return fmt::format(
+        "number of terms = {}, terms size = {}, metadata size = {}, bloom filter size = {}, dictionary size = {}, posting lists size = {}",
+        num_terms,
+        ReadableSize(current_size),
+        ReadableSize(metadata_file_size),
+        ReadableSize(bloom_filter_file_size),
+        ReadableSize(dictionary_file_size),
+        ReadableSize(posting_lists_file_size));
+}
+
+GinIndexStore::Statistics GinIndexStore::getStatistics()
+{
+    return {
```

Comment:
Personally I prefer to have a constructor instead of initializing struct members manually.

By default subclasses have access to provate members like friends, so it is easy to do a constructor:

```
explicit Statistics::Statistics(const GinIndexStore &)
{
...
}

GinIndexStore::Statistics GinIndexStore::getStatistics()
{
   return GinIndexStore::Statistics(*this);
}
```